### DISCUSSION TITLE
TLS not working - supported?

### DISCUSSION THREAD

**Discussion by MrDonCherry** [jomjol/AI-on-the-edge-device#3054] [General]
I updated the config.ini with the parameter
from the Docu 1.6.1 Parameter CACert
in the following manner
[MQTT]
Uri = mqtt://myserver:8883
MainTopic = gaszaehler
CACert = /config/certs/ca.pem

I tried 1st .crt and also .pem format for the root certificate.

In the log from my mosquitto server, I find the following entry.

1714143887: New connection from x.x.x.x:57438 on port 8883.
1714143887: OpenSSL Error[0]: error:0A00010B:SSL routines::wrong version number
1714143887: Client <unknown> disconnected: Protocol error.

Is TLS supported?

**Comment by ghost** [comment]
Sorry, is not. The main space is too small to enable ssl/tls.

**Comment by Slider0007** [comment]
@MrDonCherry 
It seems like a TLS version mismatch. TLS v1.2 should be supported, though. Check if mosquitto is acepting TLS1.2 connections.

**Reply by caco3** [reply]
@MrDonCherry and @MysticFlow Can you confirm it works with TLS 1.2?

so I can update the docs: https://github.com/jomjol/AI-on-the-edge-device/pull/3213

**Reply by caco3** [reply]
@MrDonCherry and @MysticFlow  Ping

**Comment by MysticFlow** [comment]
> I updated the config.ini with the parameter from the Docu 1.6.1 Parameter CACert in the following manner [MQTT] Uri = mqtt://myserver:8883 MainTopic = gaszaehler CACert = /config/certs/ca.pem
> 
> I tried 1st .crt and also .pem format for the root certificate.
> 
> In the log from my mosquitto server, I find the following entry.
> 
> 1714143887: New connection from x.x.x.x:57438 on port 8883. 1714143887: OpenSSL Error[0]: error:0A00010B:SSL routines::wrong version number 1714143887: Client disconnected: Protocol error.
> 
> Is TLS supported?
